### PR TITLE
feature: add example .yaml resource files which support to run Apache APISIX on kubernetes

### PR DESCRIPTION
**Pull request by totemofwolf**
### Summary

➜  kubernetes git:(k8s) tree .
.
├── README.md
├── apisix-gw-config-cm.yaml
├── deployment.yaml
├── service-aliyun-slb.yaml
└── service.yaml

### Full changelog

* add example .yaml resource files which support to run `Apache APISIX` on kubernetes

### CODE REVIEW

File: kubernetes/apisix-gw-config-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        client_body_timeout: 60s       # timeout for reading client request body, then 408 (Request Time-out) error is returned to the client
+        send_timeout: 10s              # timeout for transmitting a response to the client.then the connection is closed
+        underscores_in_headers: "on"   # default enables the use of underscores in client request header fields
+        real_ip_header: "X-Real-IP"    # http://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header
+        real_ip_from:                  # http://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from
+          - 127.0.0.1
+          - 'unix:'
+
+    etcd:
+      host: "http://127.0.0.1:2379"   # etcd address
```
Comment (nic-chen) [inline kubernetes/apisix-gw-config-cm.yaml]: Should  remind to  modify the etcd config in  README

Comment (totemofwolf) [reply kubernetes/apisix-gw-config-cm.yaml]: > Should remind to modify the etcd config in README

done

File: kubernetes/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.name
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.namespace
+          image: 'YOUR_APISIX_IMAGE_URL:latest'
```
[the remark this answers is not in the document]
Comment (totemofwolf) [reply kubernetes/deployment.yaml]: > Should set a default image here

done

File: kubernetes/README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+### Usage
+
```
Comment (membphis) [inline kubernetes/README.md]: need some basic introductions.

Comment (totemofwolf) [reply kubernetes/README.md]: > need some basic introductions.

For instance ?

Comment (membphis) [reply kubernetes/README.md]: Why do we need these default profiles? What are these configuration files for?

Comment (membphis) [reply kubernetes/README.md]: When do users need these profiles?

Comment (moonming) [reply kubernetes/README.md:21]: @membphis can we merge it first?

**Pull request merged**